broker.go

```diff
@@ -68,20 +71,29 @@ func (b awsAccountBroker) Services(ctx context.Context) []brokerapi.Service {
 	}
 }
 
+type serviceInstance struct {
+	gorm.Model
+	InstanceId string
+	RequestId  string
+}
+
 func (b awsAccountBroker) Provision(ctx context.Context, instanceID string, details brokerapi.ProvisionDetails, asyncAllowed bool) (brokerapi.ProvisionedServiceSpec, error) {
 	spec := brokerapi.ProvisionedServiceSpec{}
 	if !asyncAllowed {
 		return spec, brokerapi.ErrAsyncRequired
 	}
 
 	email := generateUniqueEmail(b.baseEmail, instanceID)
-	_, err := b.mgr.CreateAccount("Service Broker account", email)
+	createResult, err := b.mgr.CreateAccount(instanceID, email)
 	if err != nil {
 		return spec, err
 	}
-	// TODO use the result?
 
 	b.logger.Info("Account created for " + email)
+	var requestID = *createResult.CreateAccountStatus.Id
+	b.logger.Info("RequestId: " + requestID)
+
+	b.db.Create(&serviceInstance{InstanceId: instanceID, RequestId: requestID})
```

Comment:
Minor: I’d refactor the persistence out of Provision.

Reply:
I agree...just not sure where to put it.